### PR TITLE
States logic

### PR DESCRIPTION
**Pull request by joseguti84**

### CODE REVIEW

File: source/CommonJobs/CommonJobs.MVC.UI/Scripts/evaluations-dashboard.js [hunk 1]
```diff
@@ -183,8 +191,50 @@
...
+                case 0:
+                case 1:
+                case 2:
+                case 3:
+                    return "Calificar";
+                default:
+                    return "Ver Calificación";
+            }
+        }, this);
+        this.calificationActionClass = ko.computed(function () {
```
Comment (joseguti84) [inline source/CommonJobs/CommonJobs.MVC.UI/Scripts/evaluations-dashboard.js]: I need improve this code

File: source/CommonJobs/CommonJobs.MVC.UI/Scripts/evaluations-dashboard.js [hunk 2]
```diff
@@ -183,8 +191,50 @@
...
+                case 2:
+                    return "Calificar como responsable";
+                case 1:
+                case 3:
+                    return "Calificar como empresa";
+                default:
+                    return "Ver Calificación";
+            }
+        }, this);
+        this.calificationActionText = ko.computed(function () {
```
Comment (joseguti84) [inline source/CommonJobs/CommonJobs.MVC.UI/Scripts/evaluations-dashboard.js]: I need improve this code

File: source/CommonJobs/CommonJobs.MVC.UI/Scripts/evaluations-dashboard.js [hunk 3]
```diff
@@ -183,8 +191,50 @@
...
+            return (this.evaluatorsAmount() === 1) ? this.evaluatorsAmount() + " calificador" : this.evaluatorsAmount() + " calificadores";
+        }, this);
         this.state = data.State;
-        this.stateName = evaluationStates[data.State];
+        this.stateName = evaluationStates[this.state];
+        this.stateClasses = "state-doc state-" + this.state;
+        this.isCalificatorsEditable = ko.computed(function () {
+            return this.isResponsible && this.state != 6;
+        }, this);
+        this.calificationActionTooltip = ko.computed(function () {
```
Comment (joseguti84) [inline source/CommonJobs/CommonJobs.MVC.UI/Scripts/evaluations-dashboard.js]: I need improve this code

**Comment by andresmoschini** [comment]
I would refer to have text literals on views in order to helpe to understand view logic easily, but I am not sure how to do that in current scenario, so, I am fine with you approach.

File: source/CommonJobs/CommonJobs.MVC.UI/Scripts/evaluations-dashboard.js [hunk 4]
```diff
@@ -183,8 +191,39 @@
...
+            return this.isResponsible && this.state != 6;
+        }, this);
+        this.calificationActionTooltip = ko.observable('');
+        this.calificationActionText = ko.observable('');
+        this.calificationActionClass = ko.observable('');
+        this.calificationAction = ko.computed(function () {
+            switch (this.state) {
+                case 0:
+                case 2:
+                    this.calificationActionTooltip("Calificar como responsable");
```
Comment (andresmoschini) [inline source/CommonJobs/CommonJobs.MVC.UI/Scripts/evaluations-dashboard.js]: I think that it is not a good idea to have collateral effects on computed evaluation. Maybe it is better to have a subscription, or better yet, multiple computed properties.

**Comment by joseguti84** [comment]
I agreed with you. But in this case, it's the only way to do that dynamically.

File: source/CommonJobs/CommonJobs.MVC.UI/Scripts/evaluations-dashboard.js (hunk 4 above)
Comment (joseguti84) [inline source/CommonJobs/CommonJobs.MVC.UI/Scripts/evaluations-dashboard.js]: I changed this compueted to a suscribe method.

**Pull request merged**